lib/matplotlib/axes/_axes.py

```diff
@@ -1117,6 +1117,11 @@ def hlines(self, y, xmin, xmax, colors='k', linestyles='solid',
         xmin = self.convert_xunits(xmin)
         xmax = self.convert_xunits(xmax)
 
+        if colors is None and rcParams['lines.color'] is None:
```

Comment:
We ensure that `rcParams['lines.color']` is always a valid color, see

https://github.com/matplotlib/matplotlib/blob/17ef13b31339481a3a9f293ca0c91c994a924daf/lib/matplotlib/rcsetup.py#L337-L345

and 

https://github.com/matplotlib/matplotlib/blob/17ef13b31339481a3a9f293ca0c91c994a924daf/lib/matplotlib/rcsetup.py#L1047

so we do not need this check.

Reply:
@tacaswell if we remove this check, do we even need to try and default to 'k'? As within `vlines` and `hlines`, the `colors` parameter is given to construct a `LineCollection` which itself defaults to `rcParams['lines.color']`